sql/ddl_internal.sql

```diff
@@ -69,31 +54,98 @@ BEGIN
     END IF;
 
     --create time dimension
-    INSERT INTO _timescaledb_catalog.dimension(hypertable_id, column_name, column_type, aligned,
-        num_slices, partitioning_func_schema, partitioning_func,
-        interval_length
-    ) VALUES (
-        hypertable_row.id, time_column_name, time_column_type, TRUE,
-        NULL, NULL, NULL,
-        chunk_time_interval
-    );
+    PERFORM _timescaledb_internal.add_dimension(main_table,
+                                                hypertable_row,
+                                                time_column_name,
+                                                NULL,
+                                                chunk_time_interval,
+                                                FALSE);
 
     IF partitioning_column IS NOT NULL THEN
         --create space dimension
-        INSERT INTO _timescaledb_catalog.dimension(hypertable_id, column_name, column_type, aligned,
-            num_slices, partitioning_func_schema, partitioning_func,
-            interval_length
-        ) VALUES (
-            hypertable_row.id, partitioning_column, partitioning_column_type, FALSE,
-            number_partitions::smallint, partitioning_func_schema, partitioning_func,
-            NULL
-        );
+        PERFORM _timescaledb_internal.add_dimension(main_table,
+                                                    hypertable_row,
+                                                    partitioning_column,
+                                                    number_partitions,
+                                                    NULL);
     END IF;
 
     RETURN hypertable_row;
 END
 $BODY$;
 
+CREATE OR REPLACE FUNCTION _timescaledb_internal.add_dimension(
```

Comment:
Thought about this. So, dimensions are still "approx" immutable. This PR only allows you to add new dimensions, not modify existing ones. I figured that if people would want to add new dimensions after inserting first data, there's no reason to not allow that because I see no reason it wouldn't work. We could block this for now if that would make you feel more comfortable, but I doubt people will use this API until we tell them about it or start documenting it.